### DISCUSSION TITLE
Memory leak x-if

### DISCUSSION THREAD

**Discussion by antman3351** [alpinejs/alpine#3399] [7. Bugs (archived)]
Hi, 
I'm using Alpine to populate a table, I've noticed every time the table is built the memory usage increases ( especially noticeable on large tables ).

Here a simple example https://codepen.io/antman3351/pen/poZZqjd ( Only works in Chrome as FireFox doesn't have `window.performance.memory` )

The "hide"/"show" button just toggles a variable used by x-if that wraps the table, this is causing the problem. 
The "generate new data" button re-creates the array of data, curiously changing the array of data doesn't result in much of a memory increase over time.

If you want to run a profile snap-shot I recommend running it outside of codepen as it was giving me strange results.

**Comment by ekwoka** [comment]
I'm not sure if this is a just an issue of garbage collection being slower than you'd want for something like this...or if it's actually Alpine holding on to too many old references. it does seem to drop down after some time...just quite slowly.

**Comment by antman3351** [comment]
In chrome you can force garbage collection in the performance tab ( I did while testing), so I don't think the GC is the problem.
